### ISSUE TITLE
Ability to sort within an album

### ISSUE THREAD

**Issue by zeroepoch** [daneren2005/Subsonic#601]
Maybe I'm not seeing the setting, but I don't see a way to sort within an album.  For example if the tracks in the album have special names and mean something (ex: EP100, EP101) and I want to sort it by name.  Or the tracks have in the metadata (details) have a year it would be nice to sort or reverse sort the items to find the one with the latest date.  Does this make sense?

**Comment by daneren2005** [comment]
I'm not really sure if I intend on adding any custom sorting right now.  I might add it for tablets but not on phones or something.  I haven't really decided since I need to come up with a good way to do it in such a small amount of space.  What are some example sortings that you would find useful?  And keep in mind I will not be ever doing anything based off of IDv3 tags since songs may or may not be downloaded at any given point (and transcoding often removes them).  It will need to be data available in a getMusicDirectory call (ie: http://www.subsonic.org/pages/inc/api/examples/directory_example_2.xml)

**Comment by zeroepoch** [comment]
I assume the default sort order is probably track number or something?  I was trying to think of ways to edit the files (tags/names) to make it work currently such as faking out the track number with a date code like 20151222 or something.  That doesn't seem like a very good solution though.  Another option that could be easy to workaround is simply sorting by filename.  I think that may be enough to make something work.  For podcast downloads that would be fine since the filenames usually start with episode number.  We started by using the podcast feature in Subsonic but it kept downloading the same file over and over again strangely.  We've now switch to podgrab so looking for a simple way to know the latest "track" in an "album".

**Comment by daneren2005** [comment]
The more I think about it the more I don't really want to add custom sorting just for a hack work around. I understand why you had to do it but the entire point in having a dedicated podcasts section is to cover issues like this. If you are going to have something separate to download the podcasts then you probably just need to also make sure that it a is setting the track # so it sorts correctly.

**Comment by zeroepoch** [comment]
That makes sense.  We'll workaround it by updating the files.  Thanks!

**Comment by daneren2005** [comment]
Can you post an example feed that double downloads the podcasts? I'm also curious to see if I can fix the issue on the server end (though no promises)?

**Comment by zeroepoch** [comment]
It's my friend's server but I think this was the one I think he saw the issues on, http://escapepod.org/feed/

**Comment by zeroepoch** [comment]
So my friend and I found out it's a subsonic bug.  If you set it to download all the episodes from a podcast it will download all of them and then start over again and again.  Each time it adds a number suffix starting with *0.mp3 then *1.mp3, etc.  I think this is messing with D-Sub.  Anyways we've set it to check every hour now and only download the latest episode then manually downloaded the older episodes.  Now it's working fine.  Haven't retested this since upgrading to subsonic 6.0.
